packages/server/src/course/course.service.ts

```diff
@@ -234,4 +236,63 @@ export class CourseService {
       );
     }
   }
+
+  async getUserInfo(
+    courseId: number,
+    page: number,
+    pageSize: number,
+    role?: Role,
+    search?: string,
+  ): Promise<UserPartial[]> {
+    try {
+      const query = await getRepository(UserModel)
+        .createQueryBuilder()
+        .leftJoin(
+          UserCourseModel,
+          'UserCourseModel',
+          '"UserModel".id = "UserCourseModel"."userId"',
+        )
+        .where('"UserCourseModel"."courseId" = :courseId', { courseId });
+
+      // check if searching for specific role
+      if (role) {
+        query.andWhere('"UserCourseModel".role = :role', { role });
+      }
+      // check if searching for specific name
+      if (search) {
+        const likeSearch = `%${search}%`.toUpperCase();
+        query.andWhere(
+          new Brackets((q) => {
+            q.where('UPPER("UserModel"."firstName") like :firstSearch', {
+              firstSearch: likeSearch,
+            }).orWhere('UPPER("UserModel"."lastName") like :lastSearch', {
+              lastSearch: likeSearch,
+            });
+          }),
+        );
+      }
+
+      // run query
+      const users = query
+        .select([
+          'UserModel.id',
+          'UserModel.firstName',
+          'UserModel.lastName',
+          'UserModel.photoURL',
+          'UserModel.email',
+        ])
+        .orderBy('UserModel.firstName')
+        .skip((page - 1) * pageSize)
+        .take(pageSize)
+        .getMany();
+
+      return users;
+    } catch (err) {
+      console.error(err);
+      throw new HttpException(
+        ERROR_MESSAGES.common.pageOutOfBounds,
```

Comment:
To follow up on this, would this function ever throw an error? Just from a quick read, it seems like at worst it would return an empty list of users. Not 100% sure how take/skip behave, so feel free to correct me

Reply:
I just tested it and I think you're right in the sense that the page can't go out of bounds. I'm not sure if other errors are possible though. It seems like if there are any malformed query parameters, the query would just return an empty list?